### ISSUE TITLE
[Beta 14] Issues with dragging to show/hide the side menu on Android 

### ISSUE THREAD

**Issue by cbruun** [driftyco/ionic#2674]
I created a Codepen where the issue can be reproduced. Link: http://codepen.io/anon/pen/EaPjJR . It is based on the official Side Menu and Navigation Codepen (http://codepen.io/ionic/pen/tcIGK).

The only difference between the two is that I added a handle on the left side of the screen with a toggle menu directive.

On Android, If I start the side menu drag on the navigation bar or the handle, nothing happens. On desktop Chrome and on iOS this works just fine.

Also, when the side menu is open, it is not possible to use a drag to close it again. Only a click on the content area or the navicon closes it.

The problem occurs on Android 4.4.4 and 4.2.2 and possible more versions.

**Comment by beartiago** [comment]
I'm testing my app using CrossWalk 11.39.258.0 and the menu close swipe back working!
LG G3 D855P Android 4.4.2

But now is crashing on android 4.2
```01-05 12:11:16.362: E/WifiManager(12177): mWifiServiceMessenger == null
01-05 12:11:16.680: E/dalvikvm(12177): Could not find class 'android.bluetooth.BluetoothManager', referenced from method org.chromium.media.AudioManagerAndroid.hasBluetoothHeadset
01-05 12:11:17.631: E/chromium(12177): [ERROR:unix_domain_client_socket_posix.cc(101)] Not implemented reached in virtual int net::UnixDomainClientSocket::GetPeerAddress(net::IPEndPoint*) const
01-05 12:11:17.632: E/chromium(12177): [ERROR:unix_domain_client_socket_posix.cc(101)] Not implemented reached in virtual int net::UnixDomainClientSocket::GetPeerAddress(net::IPEndPoint*) const
01-05 12:11:17.633: A/libc(12177): Fatal signal 11 (SIGSEGV) at 0x07b5f852 (code=1), thread 12177 (tan.CapptanPass)```

**Comment by rafaellop** [comment]
I've tried the dirty way proposed by @bwasnie1 (above) and it worked for me on all my test devices (4.0.4, 4.2.2 and 4.4.3). Without the hack I cannot open side menu if the current page container is a div element with overflow scroll. With the hack both the side menu is working properly when dragged and the div can be scrolled.

**Comment by beartiago** [comment]
ok, solved my problem using Crosswalk 10 (:

**Comment by ajoslin** [comment]
OK, the problem is:

With native scrolling enabled on Android 4.4-, the browser will send a touchcancel as soon as scrolling starts and abort any further touchmove events.  The only way to stop this is to call e.preventDefault() on the touchmove events.

How to fix this? Well... it's going to require some though.

**Comment by adamdbradley** [comment]
Please test out this fix and let me know how it goes: https://github.com/driftyco/ionic/commit/a5881eaeba7d12d016fe3712c77261e75e863cf3

I'm going to close this issue so we can keep the discussion on https://github.com/driftyco/ionic/issues/1729

Thanks

**Comment by alzalabany** [comment]
problem still exists on nexus 5 , android 5.1.1

**Comment by aorylamballe** [comment]
Hi,

It seems the issue is still here with native scrolling: we cannot drag in the side menu (it works with JavaScript scrolling, i.e. sidemenu starter app default configuration).

Steps to reproduce:

- Install the sidemenu starter app: ```ionic start myApp sidemenu```
- Edit the config() to add native scrolling: ```$ionicConfigProvider.scrolling.jsScrolling(false);```
- Deploy on an Android device (mine: Xiaomi MIUI / Android 4.4.4)
- Try to drag in the side menu: nothing happens. Only the button allows to show the menu.

The following workaround in run() (found in another post) fixes the issue to close the menu by swiping. But removing the condition to make it work to open the menu breaks the scrolling.

```
if (ionic.Platform.isAndroid()) {
	document.addEventListener('touchstart', function (event) {
		if ($ionicSideMenuDelegate.isOpenLeft()) {
			event.preventDefault();
		}
	});
}
```

I checked the link in @adamdbradley 's last response, but I didn't find solution for that. Any idea about what could be done?

Thanks!

**Comment by seanheinen** [comment]
I've had this same issue to this day.

I managed to get mine working 100% by making a simple change to the ionSideMenuContent directive.

In the compile function property there is a function called onDragX.
By removing the if getPrimaryScrollAxis condition mine works:
<code>
function onDragX(e) {
  if (!sideMenuCtrl.isDraggableTarget(e)) return;
  //if (getPrimaryScrollAxis(e) == 'x') {  
      sideMenuCtrl._handleDrag(e);
      e.gesture.srcEvent.preventDefault();
  //}
}
</code>

I understand there is probably a reason for this check. From my understanding it to do with whether or not the user swiped an X which then turned into a Y. I've played around with the swiping since and my experience hasn't been affected negatively in any way..
